app/models/mixins/purging_mixin.rb

```diff
@@ -63,11 +65,13 @@ def purge_timer
       purge_queue(*purge_mode_and_value)
     end
 
-    def purge_queue(mode, value = nil)
+    def purge_queue(mode, value = nil, sync: false)
+      # Use sync: true to avoid contention on the same table.
       MiqQueue.submit_job(
         :class_name  => name,
         :method_name => "purge_by_#{mode}",
-        :args        => [value]
+        :args        => [value],
+        :task_id     => !!sync ? PURGE_TASK_ID : nil
```

Comment:
Note, task_id is listed as deprecated from https://github.com/ManageIQ/manageiq/pull/15240

`# @options options [String] :task_id (deprecated)`

I don't know why. I am posting here to see if there's a problem with using it.  I think the use case is valid.  I don't remember if we were planning on removing task_id from submit_job.  I guess purge_queue could directly do MiqQueue.put if needed.

Reply:
It's deprecated in the sense that we want to get rid of it so we don't want new code using it.

Reply:
Wait are we taking taskid (the serialization thing) or like miq_task_id (reference to MiqTasks)?

Reply:
lol, I flipped the sync/async wording.  Let me fix it.

task_id for async / sequential queue processing

Reply:
ok, I think it reads better with concurrent... no double negatives.

Reply:
ok, check it now.  I can use deliver_on or another mechanism but task_id does exactly what we want here.  No extra delay in processing results via deliver_on.

Reply:
This is probably fine for now. Curious what @kbrock thinks.  Part of me is thinking a callback or an actual MiqTask might be better, but it's me on begging this to fix the actual bug.